app/models/Missions.js

```diff
@@ -73,10 +74,38 @@ var MissionSchema = function(outcomeType) {
 util.inherits(MissionSchema, Schema);
 
 var missionSchema = new MissionSchema();
-missionSchema.pre('save', function(next) {
+missionSchema.pre('save', function(next){
     var that = this;
     var missionType = that.getType();
 
+    if(that.isFirstOfType === undefined){
```

Comment:
should use `typeof` to compare with undefined:
`if (typeof that.isFirstOfType === 'undefined') {`